include/sparrow/typed_array.hpp

```diff
@@ -186,7 +194,31 @@ namespace sparrow
         layout_type m_layout;
     };
 
+    template <typename T, std::size_t N>
+struct template_param_at;
+
+template <template <typename...> class L, typename A, typename... Bs, std::size_t N>
+struct template_param_at<L<A, Bs...>, N> {
+    using type = typename template_param_at<L<Bs...>, N - 1>::type;
+};
+
+template <template <typename...> class L, typename A, typename... Bs>
+struct template_param_at<L<A, Bs...>, 0> {
+    using type = A;
+};
+
+template <class L, std::size_t N>
+using template_param_at_t = typename template_param_at<L, N>::type;
+
     // Constructors
+    template <class T, class L>
```

Comment:
Could you clarify that `L` is a layout? I thought it was a type-list first. Maybe just name it `Layout`?

Reply:
Renamed Layout